web/js/notification-manager.js

```diff
@@ -17,16 +18,30 @@ function init(mainDOMElement) {
 	mainEl = mainDOMElement;
 	bannerEl = document.getElementById("notification-banner");
 	formEl = bannerEl.querySelector("form");
-	msgEl = bannerEl.querySelector(".msg");
+	messageEl = bannerEl.querySelector(".msg");
 	cancelBtnEl = bannerEl.querySelector("button.cancel-btn");
 	okBtnEl = bannerEl.querySelector("button[type=submit]");
 
 	formEl.addEventListener("submit",onFormSubmit,false);
 	cancelBtnEl.addEventListener("click",() => hide(/*result=*/false),false);
 }
 
-function show(msg,isModal = false,isError = false,showCancel = false,onClose = () => {}) {
-	closeCallback = onClose;
+/**
+ * Displays a notification to the end-user
+ * @param {string} message
+ * @param {Object} options
+ * @param {boolean} options.isModal if isModal is true, clear after a set duration
```

Comment:
Perhaps I'm misunderstanding, but "clear after a set duration" is when the notification is NOT modal, I think. In this implementation, I think "modal" means "must click a button to dismiss".

Reply:
You are correct. I'll get that updated.